frontend/form-builder/src/components/FormPreview/FormPreview.tsx

```diff
@@ -1,40 +1,73 @@
 import { Heading, Paragraph } from "@digdir/design-system-react";
-import React from "react";
-import { ComponentIcon, XMarkIcon } from "@navikt/aksel-icons";
+import React, { useContext, useEffect, useRef } from "react";
+import { ComponentIcon } from "@navikt/aksel-icons";
 import classes from "./FormPreview.module.css";
 import { useTranslation } from "react-i18next";
 import { useDrop } from "react-dnd";
 import { DraggableItemsType } from "../../types/dndTypes";
 import { FormComponent } from "../../../../main-app/src/components/FormComponent";
+import { FormBuilderContext } from "../../context";
 
 interface FormPreviewProps {
   settingsRef: React.RefObject<HTMLDialogElement>;
-  formComponents: FormComponent[];
-  setFormComponents: React.Dispatch<React.SetStateAction<FormComponent[]>>;
 }
 
-export const FormPreview = ({
-  settingsRef,
-  formComponents,
-  setFormComponents,
-}: FormPreviewProps): React.JSX.Element => {
+export const FormPreview = ({ settingsRef }: FormPreviewProps): React.JSX.Element => {
   const { t } = useTranslation();
+  const { form, setForm, setCurrentComponent } = useContext(FormBuilderContext);
+  const formRef = useRef<Form>();
+
+  useEffect(() => {
+    formRef.current = form;
+  }, [form]);
 
   const [{ isOver }, drop] = useDrop(() => ({
     accept: DraggableItemsType.ToolbarItem,
     drop: (item: FormComponent) => {
-      setFormComponents((prev) => [...prev, item]);
+      const randomString = Math.random().toString(36).substring(2, 7);
+      const updatedItem = { ...item, name: randomString };
+      const updatedComponents = [...formRef.current.components, updatedItem];
+      setForm({ ...formRef.current, components: updatedComponents });
     },
     collect: (monitor) => ({
       isOver: !!monitor.isOver(),
     }),
   }));
 
-  const handleRemoveItem = (index: number) => {
-    const newItems = formComponents.filter((_, i) => i !== index);
-    setFormComponents(newItems);
+  // TODO: Fix buggy behaviour when clicking in mobile view
+  const handleClick = (item: FormComponent, index: number) => {
+    item.order = index;
```

Comment:
Try to avoid updating objects directly in such way :) 
Use `setState` or a new variable instead.
`const updatedItem = { ...item, order: index };`

Reply:
Ok, will keep that in mind 😊
But if I do that now, the modal won't show at all.
Will fix this in #176.